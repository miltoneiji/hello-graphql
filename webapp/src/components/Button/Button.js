import React from 'react';
import styled from 'styled-components';

const colors = {
  primary: '#007bff',
  success: '#28a745',
  danger: '#dc3545',
  link: 'transparent',
};

const Button = styled.button`
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  max-width: 250px;
  border-radius: 50px;
  padding: 10px 16px;
  background-color: ${props => colors[props.styleType || 'primary']};
  color: ${props => props.styleType && props.styleType === 'link' ? '#007bff' : 'white'}
  cursor: pointer;


  // move to a theme
  font-family: sans-serif;
  font-size: 20px;
  font-weight: 400;
  line-height: 1;

  text-decoration: none !important;

  &:disabled {
    opacity: 0.65;
    cursor: default;
  }
`;

export default Button;
