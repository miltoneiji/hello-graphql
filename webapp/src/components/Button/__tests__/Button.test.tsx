import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import Button from '../Button';

describe('<Button />', () => {
  it('renders properly', () => {
    const { container } = render(
      <Button>
        Hello, paper!
      </Button>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
