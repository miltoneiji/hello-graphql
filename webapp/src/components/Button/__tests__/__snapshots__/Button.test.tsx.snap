// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders properly 1`] = `
.c0 {
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  max-width: 250px;
  border-radius: 50px;
  padding: 10px 16px;
  background-color: #007bff;
  color: white;
  cursor: pointer;
  font-family: sans-serif;
  font-size: 20px;
  font-weight: 400;
  line-height: 1;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
}

.c0:disabled {
  opacity: 0.65;
  cursor: default;
}

<button
  class="c0"
>
  Hello, paper!
</button>
`;
